app/src/main/java/com/infomaniak/mail/views/decoratedTextItemView/FolderItemView.kt

```diff
@@ -17,31 +17,38 @@
  */
 package com.infomaniak.mail.views.decoratedTextItemView
 
+import android.content.Context
+import android.util.AttributeSet
 import androidx.core.content.res.getDimensionPixelSizeOrThrow
 import androidx.core.view.marginEnd
 import com.infomaniak.lib.core.utils.context
 import com.infomaniak.lib.core.utils.setMarginsRelative
 import com.infomaniak.mail.R
-import com.infomaniak.mail.databinding.ViewDecoratedTextItemBinding
 import com.infomaniak.lib.core.R as RCore
 
-interface IndentableFolder {
+abstract class FolderItemView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : DecoratedTextItemView(context, attrs, defStyleAttr) {
 
-    val binding: ViewDecoratedTextItemBinding
-
-    fun setIndent(indent: Int, hasCollapsableFolder: Boolean = false, canBeCollapsed: Boolean = false) {
-        val totalStartMargin = computeStartMargin(hasCollapsableFolder, canBeCollapsed) + computeIndent(indent)
+    fun setIndent(indent: Int, hasCollapsableFolder: Boolean = false, folderCanCollapse: Boolean = false) {
+        val totalStartMargin = computeStartMargin(hasCollapsableFolder, folderCanCollapse) + computeIndent(indent)
         binding.itemName.apply { setMarginsRelative(start = totalStartMargin, end = marginEnd) }
     }
 
     private fun computeIndent(indent: Int) = binding.context.resources.getDimension(RCore.dimen.marginStandard).toInt() * indent
 
-    private fun computeStartMargin(hasCollapsableFolder: Boolean, canBeCollapsed: Boolean): Int = with(binding.context) {
-        return if (hasCollapsableFolder && !canBeCollapsed) {
+    private fun computeStartMargin(hasCollapsableFolder: Boolean, folderCanCollapse: Boolean): Int = with(binding.context) {
+        return if (hasCollapsableFolder && !folderCanCollapse) {
```

Comment:
No because hasCollapsableFolder tells that there are at least 1 collaspable folder, but folderCanCollapse tells if the current folder can be collapsed or not